lib/spack/spack/parse.py

```diff
@@ -143,7 +143,7 @@ def expect(self, id):
 
     def setup(self, text):
         if isinstance(text, string_types):
-            text = shlex.split(text)
+            text = shlex.split(str(text))
```

Comment:
@alalazo 

I need to evaluate if

* there are other calls to `shlex.split` which should also be affected
* invoking `str` is the appropriate way to convert unicode text. What is the expected effect of this?

Looking this up, it appears that other approaches take some care to decode/encode:

* https://stackoverflow.com/questions/14218992/shlex-split-still-not-supporting-unicode
* https://pypi.org/project/ushlex/

I likely won't get a chance to check this out in more depth until 7/8. In the meantime, if it seems obvious to you why `str` is the best approach, could you explain that?

Reply:
@scheibelp I modified all the other calls to `shlex.split`. Tbh, when I did this I didn't check the other approaches linked above. To my understanding the approach taken here is correct as:
- `shlex.split` expects strings as written [in the docs](https://docs.python.org/2.7/library/shlex.html)
-  `str` should implicitly convert to the type expected by string whatever is the Python version

That said Unicode across different Python versions is a rat's nest, so if you think there's something dubious it's probably worth double checking it. Let me know.